### PR TITLE
DRILL-1942-hygiene

### PR DESCRIPTION
**Pull request by cwestin**
- Formatting
- @Overrides
- finals
- some AutoCloseable additions needed for upcoming patches
- new isCancelled() abstract method on FragmentManager, implemented on subclasses;
Added missing new abstract method isCancelled(); needed for upcoming patches

Unit tests passed
Precommit suite passed (except for Windows)

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/vector/BitVector.java [hunk 1]
```diff
@@ -211,8 +220,11 @@ public TransferPair makeTransferPair(ValueVector to) {
 
   public void transferTo(BitVector target) {
     target.clear();
+    if (target.data != null) {
+      target.data.release();
```
Comment (jinfengni) [inline exec/java-exec/src/main/java/org/apache/drill/exec/vector/BitVector.java]: release(1) ?  

Also, do you intend to change from retain() to retain(1) across the source codes? Seems some places are still using retain(). Do a quick search will find couple of places, including BitVector.load(), VariableLengthVectors.java:load() etc.

**Comment by jinfengni** [comment]
+1

LGTM.

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentManager.java [hunk 2]
```diff
@@ -38,36 +37,43 @@
...
+
+  void cancel();
 
-  public abstract void cancel();
+  /**
+   * Find out if the FragmentManager has been cancelled.
+   *
+   * @return true if the FragmentManager has been cancelled.
+   */
+  boolean isCancelled();
```
Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentManager.java]: I don't see this method being used anywhere. Is this change intended?

Comment (jinfengni) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentManager.java]: "Added missing new abstract method isCancelled(); needed for upcoming patches"

Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentManager.java]: My bad.. I did not see that mentioned in the commit message, and I did not realize the PR was made with comments in addition to the commit message.

Comment (cwestin) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentManager.java]: Yes, this is a prerequisite for another patch that is coming.

File: exec/java-exec/src/main/java/org/apache/drill/exec/vector/BitVector.java (hunk 1 above)
Comment (cwestin) [inline exec/java-exec/src/main/java/org/apache/drill/exec/vector/BitVector.java]: Originally I did change all the release()s and retain()s to (1) in all files. But Jacques made me undo them in a few files he reviewed in another patch, and I haven't searched to see if there are new additions since I made the original changes. I think it's worth saving the function call for some low-level things (such as value vectors). What would you prefer?

Comment (jinfengni) [inline exec/java-exec/src/main/java/org/apache/drill/exec/vector/BitVector.java]: I think the current form is good. I'll merge the patch at it is.
